akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -158,6 +162,22 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
             .withEntity(Strict(ContentTypes.NoContentType, ByteString.empty))
         )
       }
+
+    }
+
+    "send settings" should {
+      abstract class SettingsSetup extends TestSetupWithoutHandshake with NetProbes with Http2FrameSending {
+        def expectSetting(expected: Setting): Unit = {
+          toNet.expectBytes(Http2Protocol.ClientConnectionPreface)
+          val event: FrameEvent = expectFrame()
```

Comment:
You added an `expectSETTINGS` which you can use here (it can also be changed to return a `SettingsFrame` directly).

Reply:
Good call. I was cherrypicking these changes from an old branch without that method.